### PR TITLE
(next-urql) - call init once browser side

### PR DESCRIPTION
**Pull request by JoviDeCroock**
## Summary

Passes down `urqlClient` in `withUrqlClient.getInitialProps`, this prevents the HOC from having to call `initUrqlClient` again.

If anyone has suggestions for a test feel free to throw them out there since `typeof window === undefined` in tests

Implements https://github.com/FormidableLabs/urql/issues/582

## Set of changes

Passes down `urqlClient` in `withUrqlClient.getInitialProps`

### CODE REVIEW

File: packages/next-urql/src/with-urql-client.tsx [hunk 1]
```diff
@@ -76,7 +76,7 @@ export function withUrqlClient(
       // getInitialProps runs on the server for initial render, and on the client for navigation.
       // We only want to run the prepass step on the server.
       if (typeof window !== 'undefined') {
-        return pageProps;
+        return { ...pageProps, urqlClient };
```
Comment (kitten) [inline packages/next-urql/src/with-urql-client.tsx]: How do we prevent `next` from trying to serialise it?

Comment (JoviDeCroock) [reply packages/next-urql/src/with-urql-client.tsx]: Is that a problem on client-only?

Comment (kitten) [reply packages/next-urql/src/with-urql-client.tsx]: I think it may be a problem on rehydration? @parkie-doo may know more

Comment (JoviDeCroock) [reply packages/next-urql/src/with-urql-client.tsx]: https://github.com/FormidableLabs/next-urql/commit/14d4bef6cf5754d11235461b1c686c1c4793d382#diff-605252a5383070a6def592e274ab50b4L124

Thought adding it there was the only thing affecting that, will investigate

Comment (parkerziegler) [reply packages/next-urql/src/with-urql-client.tsx]: I think this change will likely be ok. As I reason it:

- Initial SSR occurs. We skip this block altogether on the server. On the client, the call to `initUrqlClient` occurs and we get our client-side instance.
- Client-side navigation occurs and `getInitialProps` is run. `urqlClient` gets created and returned, and used in lieu of `initUrqlClient` getting called again.

I'll apply this patch locally and test it out on `examples/4-with-navigation` and ensure we still get the expected client configuration in all environments.

**Comment by parkerziegler** [comment]
Tested this out locally and everything looks good to me :shipit: 

Also, lol at this changesets name 😂 

image

**Pull request merged**